### DISCUSSION TITLE
서비스 비즈니스 로직 도메인으로 옮기기 (도메인모듈에 repository인터페이스 두기)

### DISCUSSION THREAD

**Discussion by waterfogSW** [Student-Center/weave-server#213] [개발 논의]
#212 
- 현재는 outbound port 때문에 서비스 계층에 비즈니스 로직이 일부 구현되어 있음
- 비즈니스 로직을 최대한 도메인 모델 내부로 집중시키기 위해 액션을 람다식으로 전달하는 방식을 사용함
- 하지만 이 방식은 복잡한 비즈니스 로직에는 적합하지 않을 수 있음
- 대안으로 repository 인터페이스를 도메인 모듈 내부에 위치시키는 방안이 있음 (도메인 주도 개발 시작하기 책 참고) <img width="536" alt="image" src="https://github.com/Student-Center/weave-server/assets/28651727/e725c321-bc16-4fb4-89e5-0ed8b6159ede">

- 우선 현재 방식으로 구현을 진행하되, 추후 전체적인 리팩토링을 통해 repository 인터페이스를 도메인으로 옮기는 방향으로 개선해 나가는 것이 좋겠음

**Comment by dojinyou** [comment]
요거 일단 펜딩-고민 좀 더 해봐야될듯
